Modules/_cursesmodule.c

```diff
@@ -716,9 +754,14 @@ PyCursesWindow_New(WINDOW *win, const char *encoding)
 static void
 PyCursesWindow_Dealloc(PyCursesWindowObject *wo)
 {
-    if (wo->win != stdscr) delwin(wo->win);
-    if (wo->encoding != NULL)
+    if (wo->win != stdscr && wo->win != NULL) {
+        delwin(wo->win);
+        wo->win = NULL;
```

Comment:
I don't think that it's useful since we free() the memory a few lines below. Same remark for `wo->encoding = NULL;`.

Reply:
I didn't really know whether `delwin(wo->win)` checks if its argument is NULL or not so I wanted to be sure (I'll check the manual more in detail tomorrow).

Reply:
FTR, from the man page:

> X/Open defines no error conditions. In this implementation
>
>   delwin 
>    returns an error if the window pointer is null, or if the window is the parent of another window. 
>    This implementation also maintains a list of windows, and checks that the pointer passed to delwin is one that it created, returning an error if it was not.

I added a comment saying that we silently ignore any error in `delwin`.